pkgs/misc/cups/drivers/hll3230cdw/default.nix

```diff
@@ -0,0 +1,41 @@
+{ stdenv, lib, fetchurl, cups, perl, dpkg }:
+
+let
+  version = "1.0.2-0";
+  src = fetchurl {
+    url = "https://download.brother.com/welcome/dlf103925/hll3230cdwpdrv-${version}.i386.deb";
+    sha256 = "9d49abc584bf22bc381510618a34107ead6ab14562b51831fefd6009947aa5a9";
+  };
+
+in
+stdenv.mkDerivation {
+
+  pname = "cups-brother-hll3230cdw";
+  inherit version;
+
+  buildInputs = [ cups perl ];
+  nativeBuildInputs = [ dpkg ];
+
+  inherit src;
+  dontUnpack = true;
+
+  installPhase = ''
+    dpkg-deb -x "$src" "$out"
+    mkdir -p "$out/usr/lib/cups/filter"
+    ln -s "$out/opt/brother/Printers/hll3230cdw/cupswrapper/brother_lpdwrapper_hll3230cdw" \
+      "$out/usr/lib/cups/filter/brother_lpdwrapper_hll3230cdw"
```

Comment:
Do we need to use usr/lib? Can't we just use lib? Also why do we need opt? Can we just copy the file into the final location straight away?

Reply:
I can't figure out a good solution here.  The distributed files in /opt are in a bit of a strange arrangement which I can't figure out.  I basically did the same thing as the other brother CUPS drivers.

Reply:
I have borrowed from other brother cups driver pkgs to deal with this issue.  In their infinite wisdom, brother uses the path in /opt to get information about which driver it is using and then to load the associated configuration.  I have patched most of this insanity like other packages, but I don't believe one can just rearrange the paths as the paths into /opt are hardcoded into resources of the binaries.  The solution involve using libredirect for file operations and sed on the output to always point into the store.